apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/links-section/link-table/cells/linked-with-cell/entity-selector.tsx

```diff
@@ -153,18 +157,68 @@ export const EntitySelector = ({
     if (!entitiesSubgraph) {
       return [];
     }
-    return [...getRoots(entitiesSubgraph)]
-      .filter(
-        (entity) =>
-          !entityIdsToFilterOut?.includes(entity.metadata.recordId.entityId) &&
-          !entity.metadata.archived,
-      )
+    const subgraphRoots = getRoots(entitiesSubgraph);
+
+    const hasLiveVersion: Record<EntityId, boolean> = {};
+
+    return subgraphRoots
+      .filter((entity) => {
+        const rootEntityId = entity.metadata.recordId.entityId;
+
+        if (entity.metadata.archived) {
+          return false;
+        }
+
+        if (entityIdsToFilterOut?.includes(rootEntityId)) {
+          return false;
+        }
+
+        if (includeDrafts) {
+          /**
+           * If we have included drafts in the query, we may have multiple roots for a single entity,
+           * in the case where it has a 'live' (non-draft) version and one or more unarchived draft updates.
+           *
+           * We only want one result, which should be the live version if it exists, or the single draft if there's no live.
+           * Entities without a live version can't have multiple drafts because there's nothing to fork multiple from.
```

Comment:
I've clarified that this is talking about a specific point in time in https://github.com/hashintel/hash/pull/4221/commits/99d0e561d5fbcc8b532a829e720a9a9056860a97

When updating a `draft` entity which doesn't have a `live` version you are creating new editions of the entity with that `draftId`. There are only two ways of creating an entity with a `draftId`:
1. Creating a new entity and making it `draft`
2. Updating a live entity and passing `draft: true` so that you create a new what I call draft 'series' (in the absence of a better name), i.e. a run of editions with the same `draftId`.

Reply:
So if there is no live version, there cannot be more than one draft series of editions (because you have no way of creating another series with a different `draftId`), and therefore only one root for any given point in time.

Reply:
Ah got it, thanks for the clarification 👍